lib/plugins/aws/package/compile/events/apiGateway/lib/hack/updateStage.js

```diff
@@ -0,0 +1,151 @@
+      .then(handleTracing)
+      .then(handleLogs)
+      .then(handleTags)
+      .then(applyUpdates);
+  },
+};
+
+function getAccountId() {
+  return this.provider.getAccountId().then((id) => this.accountId = id); // eslint-disable-line
+}
+
+function getRestApiId() {
+  return this.provider.request('APIGateway', 'getRestApis', { limit: 5000 })
+    .then((res) => {
+      if (res.items.length) {
+        const filteredRestApis = res.items.filter((api) =>
+          api.name === `${this.options.stage}-${this.serverless.service.service}`);
+        if (filteredRestApis.length) {
+          const restApi = filteredRestApis.shift();
+          return restApi.id;
+        }
+      }
+      return null;
+    })
+    .then((restApiId) => {
+      this.apiGatewayRestApiId = restApiId;
+    });
+}
+
+function getStage() {
+  const restApiId = this.apiGatewayRestApiId;
+
+  return this.provider.request('APIGateway', 'getStage', {
+    restApiId,
+    stageName: this.options.stage,
+  }).then((res) => {
+    this.apiGatewayStageState = res;
+  });
+}
+
+function handleTracing() {
+  const tracing = this.serverless.service.provider.tracing;
+  const tracingEnabled = !_.isEmpty(tracing) && tracing.apiGateway;
+
+  let operation = { op: 'replace', path: '/tracingEnabled', value: 'false' };
+  if (tracingEnabled) {
+    operation = { op: 'replace', path: '/tracingEnabled', value: 'true' };
+  }
+  this.apiGatewayStagePatchOperations.push(operation);
+}
+
+function handleLogs() {
+  const provider = this.serverless.service.provider;
+  const logs = provider.logs && provider.logs.restApi;
+  const ops = this.apiGatewayStagePatchOperations;
+
+  let dataTrace = { op: 'replace', path: '/*/*/logging/dataTrace', value: 'false' };
+  let logLevel = { op: 'replace', path: '/*/*/logging/loglevel', value: 'OFF' };
+
+  let operations = [dataTrace, logLevel];
+
+  if (logs) {
+    const service = this.serverless.service.service;
+    const stage = this.options.stage;
+    const region = this.options.region;
+    const logGroupName = `/aws/api-gateway/${service}-${stage}`;
+
+    // eslint-disable-next-line max-len
+    const destinationArn = { op: 'replace', path: '/accessLogSettings/destinationArn', value: `arn:aws:logs:${region}:${this.accountId}:log-group:${logGroupName}` };
+    // eslint-disable-next-line max-len
+    const format = { op: 'replace', path: '/accessLogSettings/format', value: 'requestId: $context.requestId, ip: $context.identity.sourceIp, caller: $context.identity.caller, user: $context.identity.user, requestTime: $context.requestTime, httpMethod: $context.httpMethod, resourcePath: $context.resourcePath, status: $context.status, protocol: $context.protocol, responseLength: $context.responseLength' };
+    dataTrace = { op: 'replace', path: '/*/*/logging/dataTrace', value: 'true' };
+    logLevel = { op: 'replace', path: '/*/*/logging/loglevel', value: 'INFO' };
+    operations = [destinationArn, format, dataTrace, logLevel];
+  }
```

Comment:
Same here.

(tehcnically it'll be good to incorporate Prettier, to avoid such exchanges, but of cource that's not a concern of this PR)